hilti/runtime/include/logging.h

```diff
@@ -42,39 +42,39 @@ namespace debug {
 
 namespace detail {
 /** Prints a debug message to a specific debug stream. */
-inline void print(const std::string& stream, const char* msg) {
+inline void print(std::string_view stream, const char* msg) {
     if ( ::hilti::rt::detail::globalState()->debug_logger )
         ::hilti::rt::detail::globalState()->debug_logger->print(stream, msg);
 }
 
 /** Print a string to a specific debug stream with proper escaping. */
-inline void print(const std::string& stream, const std::string_view& s) {
+inline void print(std::string_view stream, const std::string_view& s) {
```

Comment:
General question: The new `string_view` are not const, does it help in any way to make them const where we can? Plus, the 2nd argument `s` here remains a `const` `string_view`. If generally use non-const, adapt for consistency.

Reply:
Re:constness, I'd be surprised since we already pass by value so it should make no difference, but I'll check. In general `std::string_view` is designed to be very cheap to construct and copy and if used with literals often leads to the literal put into the `.TEXT` segment, e.g., for the following code both Clang and GCC seem to put `abc` into `.TEXT`.

```cpp
int foo() {
    std::string_view x{"abc"};
    puts(x.data());
}
```

The second argument was an oversight, will fix (a guideline is that `std::string_view` should always be passed by copy).

Reply:
I checked the performance of passing both args to this function as `const` values, and like I suspected it makes no difference. Checking the assembly both should produce exactly the same code.